File: 0530-minimum-absolute-difference-in-bst/0530-minimum-absolute-difference-in-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution { 
    private int minAbsDiff;
    private TreeNode prevNode;

    public int getMinimumDifference(TreeNode root) {
        minAbsDiff = Integer.MAX_VALUE;
        prevNode = null;

        inOrderTraversal(root);

        return minAbsDiff;
    }

    private void inOrderTraversal(TreeNode node) {
        if (node == null) {
            return;
        }

        inOrderTraversal(node.left);

        if (prevNode != null) {
            minAbsDiff = Math.min(minAbsDiff, node.val - prevNode.val);
        }

        prevNode = node;

        inOrderTraversal(node.right);
    }
}

File: 0530-minimum-absolute-difference-in-bst/0530-minimum-absolute-difference-in-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def getMinimumDifference(self, root: Optional[TreeNode]) -> int:
        minAbsDiff = float("inf")
        prevNode = None

        def inOrderTraversal(node):
            nonlocal minAbsDiff, prevNode
            if node is None:
                return
            inOrderTraversal(node.left)
            if prevNode is not None:
                minAbsDiff = min(minAbsDiff, node.val - prevNode.val)
            prevNode = node
            inOrderTraversal(node.right)

        inOrderTraversal(root)
        return minAbsDiff